CMakeLists.txt

```diff
@@ -16,6 +16,11 @@ if(APPLE)
    set(CMAKE_MACOSX_RPATH ON)
 endif()
 
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  list(APPEND uvwasi_defines _GNU_SOURCE _POSIX_C_SOURCE=200112)
+  list(APPEND uvwasi_sources src/unix/linux-core.c)
```

Comment:
Is this line being added by mistake?

Reply:
@addaleax , Thanks. That was a mistake. I meant to add `uvwasi.c`, based on my understanding about CMAKE that it will be subjected for `_GNU_SOURCE` macro definition. I corrected that now.

Reply:
You don't need to specify the source file, that's already done elsewhere. Take a look at libuv's CMakeList.txt, specifically the `target_compile_definitions` stanza. You need to add one for the uvwasi_a target.